pkg/xds/sync/dataplane_proxy_builder.go

```diff
@@ -136,6 +141,28 @@ func (p *DataplaneProxyBuilder) resolveVIPOutbounds(meshContext xds_context.Mesh
 					continue
 				}
 			}
+		} else {
+			if len(reachableServices) != 0 && reachableBackends == nil {
+				continue
+			}
+			if reachableBackends != nil {
+				backendKey := xds_context.BackendKey{
+					Kind: outbound.BackendRef.Kind,
+					Name: outbound.BackendRef.Name,
+					Port: outbound.BackendRef.Port,
+				}
+				// check if there is an entry with specific port or without port
+				if !pointer.Deref(reachableBackends)[backendKey] && !pointer.Deref(reachableBackends)[xds_context.BackendKey{Kind: outbound.BackendRef.Kind, Name: outbound.BackendRef.Name}] {
+					// ignore VIP outbound if reachableServices is defined and not specified
+					// Reachable services takes precedence over reachable services graph.
+					continue
+				}
+			} else if outbound.BackendRef.Kind != "MeshExternalService" {
```

Comment:
Yes, but why do we skip MeshExternalService, add a reason in the comment which is that we don't have MTP yet for MEshExternalServices if I understand the code correctly